docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+
+::
+
+    npm install https://github.com/opensearch-project/opensearch-js
+
+Next add an empty ``index.js`` file into the project and open it with your favourite editor. We’ll be adding code into this file and running the methods from the command line.
+
+Start by adding the following lines of code to connect to the cluster. Use the ``service_uri`` you’ve got in the previous section.
+
+.. code:: javascript
+
+    const { Client } = require('@opensearch-project/opensearch')
+
+    const client = new Client({
+       node: "your-open-search-service-uri"
+    })
+
+Next let's add a couple of helper methods to parse and log response body. In the code snippets we'll keep error handling somewhat simple and will use console.logs to print information into the terminal.
+
+.. code:: javascript
+
+    const logBody = (error, result) => {
+        if (error) {
+            console.error(error);
+        } else {
+            console.log(result.body);
+        }
+    }
+
+    const logTitles = (error, result) => {
+        if (error) {
+            console.error(error);
+        } else {
+            const hits = result.body.hits.hits;
+            console.log(`Number of returned results is ${hits.length}`)
+            console.log(hits.map(hit => hit._source.title));
+        }
+    }
+
+Let’s verify that we can connect to the cluster by getting all indices.
+Add the following method to index.js.
+
+.. code:: javascript
+
+    /**
+     * Retrieves and logs all present indices in the cluster.
+     */
+    module.exports.getExistingIndices = () => {
+        console.log(`Retrieving existing indices:`)
+        client.cat.indices({format: 'json'}, logResultBody)
+    };
+
+Let’s call this method from the command line. To keep commands shorter I’ll be using `run-func <https://github.com/DVLP/run-func#readme>`_, which you can install with
+
+::
+
+    npm i -g run-func
+
+To use run-func specify name of file, name of function and parameters separated with spaces.
+
+::
+
+    run-func index.js getExistingIndices
+
+If you don’t want to use an additional library, you can execute the script directly with node command:
+
+::
+
+    node -e 'require("./index").getExistingIndices()'
+
+
+`getExistingIndices` prints out a list of indices present in our cluster. Since we've just created a cluster the only index present there is ``.kibana_1`` (your name might differ), an internal index used to maintain backups when upgrading or migrating Dashboards.
+
+Getting data
+------------
+
+Now that we have our cluster running, it is time to get some data for our experiments!
+
+We’ll use a dataset from `Kaggle <https://www.kaggle.com/>`_ -  *Epicurious - Recipes with Rating and Nutrition*. It contains over 20k recipes and is perfect for data exploration! Download `full_format_recipes.json <https://www.kaggle.com/hugodarwood/epirecipes?select=full_format_recipes.json>`_, unzip and put it into the project folder.
+
+Indexing data
```

Comment:
I'm still sure that we are loading data rather than indexing it, not sure about this title

Reply:
From what I can see in docs and articles loading/ingesting/indexing are all used interchangeably. But technically we call "index" operation, that's why I called it indexing. I'm renaming the subtitle.